lib/customerio/base_client.rb

```diff
@@ -0,0 +1,76 @@
+require 'net/http'
+require 'multi_json'
+
+module Customerio
+  DEFAULT_TIMEOUT  = 10
+
+  class BaseClient
+    def initialize(auth, options = {})
+      @auth = auth
+      @timeout = options[:timeout] || DEFAULT_TIMEOUT
+
+      @json = options.has_key?(:json) ? options[:json] : true
+      @base_uri = options[:base_uri] || DEFAULT_BASE_URI
+      @timeout = options[:timeout] || DEFAULT_TIMEOUT
+    end
+
+    def request(method, path, body = nil, headers = {})
+      execute(method, path, body, headers)
+    end
+
+    private
+
+    def extract_attributes(args)
```

Comment:
only used by `track` method and is duplicated in `Client` class as well, can likely clean up one instance.